code/modules/power/gourmonger.dm

```diff
@@ -0,0 +1,276 @@
+#define GOURMONGER_METABOLISM 1
+#define GOURMONGER_STARVING 50
+#define GOURMONGER_SATISFIED 500
+#define TOO_MANY_GOURS 30
+
+var/global/gourmonger_saturation = 0
+
+/mob/living/simple_animal/hostile/gourmonger
+	name = "Gourmonger"
+	desc = "A bio-engineered lifeform designed to recycle excess organic matter into fuel."
+	icon = 'icons/mob/gourmonger.dmi'
+	icon_state = "gourmonger"
+	icon_living = "gourmonger"
+	icon_dead = "gourmonger"	//This shouldn't happen
+	faction = "gourmonger"
+	health = 125
+	maxHealth = 125
+	melee_damage_lower = 5
+	melee_damage_upper = 10	//The rads are the real danger
+	stat_attack = 2
+	search_objects = 1
+	environment_smash_flags = SMASH_CONTAINERS
+	wanted_objects = list(/obj/item/weapon/reagent_containers/food/snacks)
+	min_oxy = 0	//I dunno it breathes food or something
+	max_co2 = 0
+	var/hangry = FALSE
+	var/kcalPower = 75
+	var/growToSplit = 15
+	var/currentlyMunching = FALSE
+	var/mob/living/sniffTarget = null	//because using hostile's target var led to too many issues
+
+/mob/living/simple_animal/hostile/gourmonger/New()
+	..()
+	gourmonger_saturation++
+	growToSplit += rand(0, 15)	//For funsies
+	kcalPower += rand(0, 50)
+
+/mob/living/simple_animal/hostile/gourmonger/Life()
+	if(..())
+		if(kcalPower > 0)
+			kcalPower -= GOURMONGER_METABOLISM
+		radBurst(kcalPower/3)
+		hungerCheck()
+		if(hangry && !target)
+			if(sniffTarget)
+				chargeToPrey()
+			else
+				findPrey()
+
+/mob/living/simple_animal/hostile/gourmonger/death()
+	gourmonger_saturation--
+	radBurst(kcalPower*3)
+	divideMeat()
+	gibs(get_turf(src))
+	..()
+	qdel(src)
+
+
+/mob/living/simple_animal/hostile/gourmonger/CanAttack(var/atom/the_target) //My hands are tied. I swear I had a reason for this copy paste. It also trims some weight from targeting.
+	if(currentlyMunching)
+		return 0
+	if(the_target.invisibility)
+		return 0
+	if(isliving(the_target))
+		var/mob/living/L = the_target
+		if(L.flags & INVULNERABLE)
+			return 0
+		if(isMoMMI(L))
+			return 0
+		if(L.faction == src.faction || faction == "\ref[L]")
+			return 0
+		if(faction == "slimesummon")
+			if(isslime(L))
+				return 0
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				if((isslimeperson(H)) || (isgolem(H)))
+					return 0
+		if((istype(L,/mob/living/simple_animal/corgi/Ian) || istype(L,/mob/living/carbon/human/dummy)) && (faction == "adminbus mob"))
```

Comment:
Strange that it wouldn't attack Ian...

Reply:
That line is part of the copy paste from hostile targeting. I thought I was just misunderstanding how || and && would interact there, since the comment implies the intention is to have only adminbus mobs not attack Ian. Really I'd prefer to just get rid of that whole check since adminbus isn't even used, ever.